### PR TITLE
feat: Focus to right panel when switching between panels (ACC-295)

### PR DESCRIPTION
**Pull request by thisisamir98**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/ACC-295" title="ACC-295" target="_blank">SubtaskACC-295</a>  Focus switch between panels
  </td></table>
  <br />
 

auto focus to first focusable element of panel content when switching between preferences panels

### CODE REVIEW

File: src/script/conversation/ConversationRepository.ts [hunk 1]
```diff
@@ -2843,7 +2843,8 @@ export class ConversationRepository {
     }
 
     if (eventJson.data.status === CONVERSATION_TYPING.STOPPED) {
-      removeTypingUser(typingUser);
+      console.info(removeTypingUser);
+      // (typingUser);
```
Comment (atomrc) [inline src/script/conversation/ConversationRepository.ts]: 🤔

File: src/script/page/LeftSidebar/panels/Preferences.tsx [hunk 2]
```diff
@@ -116,7 +116,7 @@ const Preferences: React.FC<PreferencesProps> = ({contentViewModel, teamReposito
 
   return (
     <ListWrapper id="preferences" header={t('preferencesHeadline')} onClose={onClose}>
-      <ul className="left-list-items no-scroll preferences-list-items">
+      <ul role="tablist" aria-label="Prefrences tabs" className="left-list-items no-scroll preferences-list-items">
```
Comment (atomrc) [inline src/script/page/LeftSidebar/panels/Preferences.tsx]: This probably needs a copy that will be translated for the label

File: src/script/page/MainContent/panels/preferences/accountPreferences/AvatarInput.tsx [hunk 3]
```diff
@@ -43,7 +43,7 @@ const FILE_TYPES = ['image/bmp', 'image/jpeg', 'image/jpg', 'image/png', '.jpg-l
 const logger = getLogger('AvatarInput');
 
 const AvatarInput: React.FC<AvatarInputProps> = ({selfUser, isActivatedAccount, userRepository}) => {
-  const inputRef = React.useRef(null);
+  const inputRef = React.useRef<HTMLInputElement>(null);
```
Comment (przemvs) [inline src/script/page/MainContent/panels/preferences/accountPreferences/AvatarInput.tsx:46]: You not miss HTMLInputElement | null ? 🤔

File: src/script/conversation/ConversationRepository.ts (hunk 1 above)
Comment (thisisamir98) [reply src/script/conversation/ConversationRepository.ts]: my bad. fixed.

File: src/script/page/MainContent/panels/preferences/components/PreferencesPage.tsx [hunk 4]
```diff
@@ -41,8 +41,20 @@ const PreferencesPage: React.FC<PreferencesPageProps> = ({title, children}) => {
 
   const goHome = () => root?.content.switchPreviousContent();
 
+  const containerRef = useCallback((element: HTMLDivElement | null) => {
+    const nextElementToFocus = element?.querySelectorAll("[tabindex='0']")[0] as HTMLElement;
```
Comment (atomrc) [inline src/script/page/MainContent/panels/preferences/components/PreferencesPage.tsx:45]: doesn't it always selects the first one?

**Review by atomrc — CHANGES_REQUESTED** [review]
A few things to consider I believe

File: src/script/page/MainContent/panels/preferences/components/PreferencesPage.tsx (hunk 4 above)
Comment (przemvs) [inline src/script/page/MainContent/panels/preferences/components/PreferencesPage.tsx:45]: Hmm.. Maybe prepare TypeGuard for HTML Element? 🤔

File: src/script/page/MainContent/panels/preferences/accountPreferences/AvatarInput.tsx (hunk 3 above)
Comment (thisisamir98) [reply src/script/page/MainContent/panels/preferences/accountPreferences/AvatarInput.tsx:46]: image
TS detection is fine

File: src/script/page/LeftSidebar/panels/Preferences.tsx (hunk 2 above)
Comment (thisisamir98) [reply src/script/page/LeftSidebar/panels/Preferences.tsx]: done

**Review by aweiss-dev — APPROVED** [review]

**Review by atomrc — APPROVED** [review]

**Pull request merged**